src/node_os.cc

```diff
@@ -128,30 +129,32 @@ static void GetCPUInfo(const FunctionCallbackInfo<Value>& args) {
   if (err)
     return;
 
-  Local<Array> cpus = Array::New(env->isolate());
+  CHECK(args[0]->IsFunction());
+  Local<Function> genfn = args[0].As<Function>();
+
+  CHECK(args[1]->IsFloat64Array());
+  Local<Float64Array> array = args[1].As<Float64Array>();
+  CHECK_EQ(array->Length(), 6);
+  Local<ArrayBuffer> ab = array->Buffer();
+  double* fields = static_cast<double*>(ab->GetContents().Data());
+
+  CHECK(args[2]->IsArray());
+  Local<Array> cpus = args[2].As<Array>();
+
   for (i = 0; i < count; i++) {
     uv_cpu_info_t* ci = cpu_infos + i;
 
-    Local<Object> times_info = Object::New(env->isolate());
-    times_info->Set(env->user_string(),
-                    Number::New(env->isolate(), ci->cpu_times.user));
-    times_info->Set(env->nice_string(),
-                    Number::New(env->isolate(), ci->cpu_times.nice));
-    times_info->Set(env->sys_string(),
-                    Number::New(env->isolate(), ci->cpu_times.sys));
-    times_info->Set(env->idle_string(),
-                    Number::New(env->isolate(), ci->cpu_times.idle));
-    times_info->Set(env->irq_string(),
-                    Number::New(env->isolate(), ci->cpu_times.irq));
-
-    Local<Object> cpu_info = Object::New(env->isolate());
-    cpu_info->Set(env->model_string(),
-                  OneByteString(env->isolate(), ci->model));
-    cpu_info->Set(env->speed_string(),
-                  Number::New(env->isolate(), ci->speed));
-    cpu_info->Set(env->times_string(), times_info);
-
-    (*cpus)->Set(i, cpu_info);
+    fields[0] = ci->speed;
+    fields[1] = ci->cpu_times.user;
+    fields[2] = ci->cpu_times.nice;
+    fields[3] = ci->cpu_times.sys;
+    fields[4] = ci->cpu_times.idle;
+    fields[5] = ci->cpu_times.irq;
+    Local<Value> argv[] = {
+      OneByteString(env->isolate(), ci->model)
```

Comment:
I just tried that in addition to the changes I just made and did not see any difference.